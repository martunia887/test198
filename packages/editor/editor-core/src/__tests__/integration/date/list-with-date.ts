import { BrowserTestCase } from '@atlaskit/webdriver-runner/runner';
import Page from '@atlaskit/webdriver-runner/wd-wrapper';

import {
  editable,
  getDocFromElement,
  fullpage,
  quickInsert,
} from '../_helpers';

// TODO: safari keys do not work after upgrade
BrowserTestCase(
  'status.ts: Insert date into list, continue, then click back on date',
  { skip: ['ie', 'safari'] },
  async (client: any, testName: string) => {
    const browser = new Page(client);

    await browser.goto(fullpage.path);
    await browser.waitForSelector(editable);
    await browser.click(editable);

    await browser.type(editable, '* abc ');
    await quickInsert(browser, 'Date');
    await browser.waitForSelector(`[aria-label="calendar"]`);
    await browser.click(editable);
    await browser.waitForSelector(`[aria-label="calendar"]`, {}, true); // removed.
    await browser.type(editable, 'def');

    // Click status, and wait for popup
    await browser.click('.dateView-content-wrap');
    await browser.waitForSelector(`[aria-label="calendar"]`);
    await browser.click(editable);

    const doc = await browser.$eval(editable, getDocFromElement);
    expect(doc).toMatchCustomDocSnapshot(testName);
  },
);
