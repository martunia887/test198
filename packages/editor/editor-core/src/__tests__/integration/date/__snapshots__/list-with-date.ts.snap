// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`status.ts: Insert date into list, continue, then click back on date 1`] = `
Object {
  "content": Array [
    Object {
      "content": Array [
        Object {
          "content": Array [
            Object {
              "content": Array [
                Object {
                  "text": "abc ",
                  "type": "text",
                },
                Object {
                  "attrs": Object {
                    "timestamp": "1557187200000",
                  },
                  "type": "date",
                },
                Object {
                  "text": " def",
                  "type": "text",
                },
              ],
              "type": "paragraph",
            },
          ],
          "type": "listItem",
        },
      ],
      "type": "bulletList",
    },
  ],
  "type": "doc",
}
`;
